### PR TITLE
added optimal path to stuck point, and added SPL benchmarking publish…

### PR DESCRIPTION
**Pull request by Yamada-Yoshifumi**
…ed to /spl_value

### CODE REVIEW

**Comment by jl-wang6** [comment]
looks good

**Comment by kaicao233** [comment]
LGTM
**Edit**: Just noticed there seems to be lacking something.

File: snake_robot/trajectory_recording.py [hunk 1]
```diff
@@ -322,6 +351,13 @@ def goal_reached_callback(self):
 
             self.fig.canvas.draw()
             self.fig.canvas.flush_events()
+            actual_length = 0
+            optimal_length = 0
+            for i,point in enumerate(shrunk_adjusted_optimal_path[:-1]):
+                optimal_length += np.linalg.norm(shrunk_adjusted_optimal_path[i] - shrunk_adjusted_optimal_path[i+1])
```
Comment (kaicao233) [inline snake_robot/trajectory_recording.py]: Should here be ```np.array(shrunk_adjusted_optimal_path[i])``` and ```np.array(shrunk_adjusted_optimal_path[i+1])```?

File: snake_robot/trajectory_recording.py [hunk 2]
```diff
@@ -322,6 +351,13 @@ def goal_reached_callback(self):
...
+            for i,point in enumerate(self.path_points[:-1]):
+                actual_length += np.linalg.norm(self.path_points[i] - self.path_points[i+1])
```
Comment (kaicao233) [inline snake_robot/trajectory_recording.py]: Same here

**Review by kaicao233 — CHANGES_REQUESTED** [review]

File: snake_robot/trajectory_recording.py (hunk 1 above)
Comment (Yamada-Yoshifumi) [reply snake_robot/trajectory_recording.py]: Updated, thanks

**Pull request merged**